### PR TITLE
[3.x] Remove unnecessary uses of deprecated strftime()

### LINKED ISSUE #15988: [3.0.0-rc1] Deprecated warnings in setup
### Summary

During installation, after passing the database check, an error is shown:

_(link of the page where the error is shown https://url.com/setup/?action=database)_

```
Deprecated: Return type of xPDO\xPDOMap::offsetExists($offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /home/public_html/core/vendor/xpdo/xpdo/src/xPDO/xPDOMap.php on line 33 
Deprecated: Function strftime() is deprecated in /home/public_html/setup/includes/modinstall.class.php on line 479
Deprecated: Function strftime() is deprecated in /home/public_html/setup/includes/modinstall.class.php on line 203
Warning: Cannot modify header information - headers already sent by (output started at /home/public_html/core/vendor/xpdo/xpdo/src/xPDO/xPDOMap.php:16) in /home/public_html/setup/includes/request/modinstallrequest.class.php on line 228 
```

### Environment
MODX version 3.0.0-rc1
I tried to install both on PHP 7.4 and on 8.1

Download the installer from [here](https://modx.com/download/other-versions?releaseKey=revolution-3.0.0-rc1).

### PR DESCRIPTION
**Pull request by JoshuaLuckers**
### What does it do?
Replace the function `strftime` with `date` where possible. 
If the `format` paramater comes from a variable the `strftime` function is not replaced since this means a breaking change.

### Why is it needed?
In PHP 8.1 the function `strftime` is deprecated.

### How to test
Use PHP 8.1 and see a lot less deprecation warnings.

### Related issue(s)/PR(s)
Resolves #15988
Related to #15864

### CODE REVIEW

**Review by Ibochkarev — APPROVED** [review]

File: core/src/Revolution/Processors/Resource/Event/UpdateFromGrid.php [hunk 1]
```diff
@@ -65,12 +65,12 @@ public function validate()
     {
         $publishDate = $this->getProperty('pub_date');
         if (!empty($publishDate)) {
-            $this->setProperty('pub_date', strftime('%Y-%m-%d %H:%M', strtotime($publishDate)));
+            $this->setProperty('pub_date', date('Y-m-d H:i:s', strtotime($publishDate)));
```
Comment (theboxer) [inline core/src/Revolution/Processors/Resource/Event/UpdateFromGrid.php]: Y-m-d H:i

File: core/src/Revolution/Processors/Resource/Event/UpdateFromGrid.php [hunk 2]
```diff
@@ -65,12 +65,12 @@ public function validate()
...
         }
 
         $unPublishDate = $this->getProperty('unpub_date');
         if (!empty($unPublishDate)) {
-            $this->setProperty('unpub_date', strftime('%Y-%m-%d %H:%M', strtotime($unPublishDate)));
+            $this->setProperty('unpub_date', date('Y-m-d H:i:s', strtotime($unPublishDate)));
```
Comment (theboxer) [inline core/src/Revolution/Processors/Resource/Event/UpdateFromGrid.php]: Y-m-d H:i

File: core/src/Revolution/Processors/System/Info.php [hunk 3]
```diff
@@ -58,7 +58,7 @@ public function process()
         $data['database_name'] = trim($this->modx->getOption('dbname'),
             $this->modx->_escapeCharOpen . $this->modx->_escapeCharClose);
         $data['database_server'] = $this->modx->getOption('host');
-        $data['now'] = strftime('%b %d, %Y %I:%M %p', time());
+        $data['now'] = date('M d, Y H:m', time());
```
Comment (theboxer) [inline core/src/Revolution/Processors/System/Info.php]: M d, Y h:i A

File: core/src/Revolution/Registry/modDbRegister.php [hunk 4]
```diff
@@ -295,7 +295,7 @@ public function send($topic, $message, array $options = [])
                                 if (!is_int($msgIdx)) {
                                     $msgKey = $msgIdx;
                                 } else {
-                                    $msgKey = strftime('%Y%m%dT%H%M%S', $timestamp) . '-' . sprintf("%03d", $msgIdx);
+                                    $msgKey = date('YmdTHms', $timestamp) . '-' . sprintf("%03d", $msgIdx);
```
Comment (theboxer) [inline core/src/Revolution/Registry/modDbRegister.php]: Ymd\THis

File: core/src/Revolution/Registry/modFileRegister.php [hunk 5]
```diff
@@ -287,7 +287,7 @@ public function send($topic, $message, array $options = [])
                                 }
                                 $msgKey = $msgIdx;
                             } else {
-                                $msgKey = strftime('%Y%m%dT%H%M%S', $timestamp) . '-' . sprintf("%03d", $msgIdx);
+                                $msgKey = date('YmdTHms', $timestamp) . '-' . sprintf("%03d", $msgIdx);
```
Comment (theboxer) [inline core/src/Revolution/Registry/modFileRegister.php]: Ymd\THis

File: setup/includes/modinstall.class.php [hunk 6]
```diff
@@ -200,7 +200,7 @@ public function getConnection($mode = 0) {
                 [
                 'target' => 'FILE',
                 'options' => [
-                    'filename' => 'install.' . MODX_CONFIG_KEY . '.' . strftime('%Y-%m-%dT%H.%M.%S').'.log'
+                    'filename' => 'install.' . MODX_CONFIG_KEY . '.' . date('YmdHis').'.log'
```
Comment (theboxer) [inline setup/includes/modinstall.class.php]: Y-m-d\TH.i.s

File: setup/includes/modinstall.class.php [hunk 7]
```diff
@@ -476,7 +476,7 @@ public function _connect($dsn, $user = '', $password = '', $prefix = '', array $
                 [
                 'target' => 'FILE',
                 'options' => [
-                    'filename' => 'install.' . MODX_CONFIG_KEY . '.' . strftime('%Y%m%dT%H%M%S') . '.log'
+                    'filename' => 'install.' . MODX_CONFIG_KEY . '.' . date('YmdHis') . '.log'
```
Comment (theboxer) [inline setup/includes/modinstall.class.php]: Ymd\THis

File: setup/includes/modinstall.class.php [hunk 8]
```diff
@@ -509,7 +509,7 @@ private function _modx(array & $errors) {
                     [
                     'target' => 'FILE',
                     'options' => [
-                        'filename' => 'install.' . MODX_CONFIG_KEY . '.' . strftime('%Y%m%dT%H%M%S') . '.log'
+                        'filename' => 'install.' . MODX_CONFIG_KEY . '.' . date('YmdHis') . '.log'
```
Comment (theboxer) [inline setup/includes/modinstall.class.php]: Ymd\THis

File: _build/test/properties.sample.inc.php [hunk 9]
```diff
@@ -19,7 +19,7 @@
 use xPDO\xPDO;
 
 /* define some properties */
-$properties['runtime'] = strftime("%Y%m%dT%H%M%S");
+$properties['runtime'] = date('YmdHis');
```
Comment (theboxer) [inline _build/test/properties.sample.inc.php]: Ymd\THis

File: core/src/Revolution/Processors/Security/Profile/Get.php [hunk 10]
```diff
@@ -71,12 +71,12 @@ public function process()
             $userArray = array_merge($profile->toArray(), $userArray);
         }
 
-        $userArray['dob'] = !empty($userArray['dob']) ? strftime('%m/%d/%Y', $userArray['dob']) : '';
-        $userArray['blockeduntil'] = !empty($userArray['blockeduntil']) ? strftime('%m/%d/%Y %I:%M %p',
+        $userArray['dob'] = !empty($userArray['dob']) ? date('m/d/Y', $userArray['dob']) : '';
+        $userArray['blockeduntil'] = !empty($userArray['blockeduntil']) ? date('m/d/Y H:i',
```
Comment (theboxer) [inline core/src/Revolution/Processors/Security/Profile/Get.php]: m/d/Y h:i A

File: core/src/Revolution/Processors/Security/Profile/Get.php [hunk 11]
```diff
@@ -71,12 +71,12 @@ public function process()
...
             $userArray['blockeduntil']) : '';
-        $userArray['blockedafter'] = !empty($userArray['blockedafter']) ? strftime('%m/%d/%Y %I:%M %p',
+        $userArray['blockedafter'] = !empty($userArray['blockedafter']) ? date('m/d/Y H:i',
```
Comment (theboxer) [inline core/src/Revolution/Processors/Security/Profile/Get.php]: m/d/Y h:i A

File: core/src/Revolution/Processors/System/Info.php [hunk 12]
```diff
@@ -40,8 +40,8 @@ public function process()
         /* general */
         $data['modx_version'] = $this->modx->version['full_appname'];
         $data['code_name'] = $this->modx->version['code_name'];
-        $data['servertime'] = strftime('%I:%M:%S %p', time());
-        $data['localtime'] = strftime('%I:%M:%S %p', time() + $serverOffset);
+        $data['servertime'] = date('H:m:s', time());
```
Comment (theboxer) [inline core/src/Revolution/Processors/System/Info.php]: h:i:s A

File: core/src/Revolution/Processors/System/Info.php [hunk 13]
```diff
@@ -40,8 +40,8 @@ public function process()
...
+        $data['localtime'] = date('H:m:s', time() + $serverOffset);
```
Comment (theboxer) [inline core/src/Revolution/Processors/System/Info.php]: h:i:s A

**Review by theboxer — CHANGES_REQUESTED** [review]

**Review by theboxer — APPROVED** [review]

**Comment by JoshuaLuckers** [comment]
@theboxer thank you!

**Comment by Ruslan-Aleev** [comment]
Why is the time format different in some files? Maybe it's worth bringing to the same format (where it possible)?

**Pull request merged**